src/utils/daemon.rs

```diff
@@ -44,80 +45,159 @@ pub fn start_daemon() {
         exit(0);
     }
 
-
     // check new crates every minute
     thread::spawn(move || {
+        // space this out to prevent it from clashing against the queue-builder thread on launch
+        thread::sleep(Duration::from_secs(30));
         loop {
+            let opts = opts();
+            let mut doc_builder = DocBuilder::new(opts);
+
+            debug!("Checking new crates");
+            match doc_builder.get_new_crates() {
+                Ok(n) => debug!("{} crates added to queue", n),
+                Err(e) => error!("Failed to get new crates: {}", e),
+            }
+
             thread::sleep(Duration::from_secs(60));
+        }
+    });
+
+    // build new crates every minute
+    thread::spawn(move || {
+        let opts = opts();
+        let mut doc_builder = DocBuilder::new(opts);
+
+        /// Represents the current state of the builder thread.
+        enum BuilderState {
```

Comment:
When i initially wrote this enum, i had thought that the logic would be more complicated than it wound up being. This can probably be reduced to an `Option<usize>`, but i left this state enum in to make the usage code easier to read.

Reply:
Update: I tried to reduce this to a `usize` and then to an `Option<usize>`, and proceeded to introduce bugs into the implementation. I think i'll keep this state enum as-is, since it provides a more readable implementation, IMO. (It'll take up the same space on the stack as an `Option<usize>` regardless.)